<a href="https://colab.research.google.com/github/AureliaWambui19/EMAIL-SPAM-DETECTION-NAYES-BAYES/blob/main/aurelia_wambui_week_9_assignment2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1. Introduction

- Email spam, also referred to as junk email, is unsolicited messages sent in bulk by email (spamming).
- The name comes from a Monty Python sketch in which Spam is ubiquitous, unavoidable, and repetitive.[1] Email spam has steadily grown since the early 1990s, and by 2014 was estimated to account for around 90% of total email traffic.[2]
-Most email spam messages are commercial in nature. Whether commercial or not, many are not only annoying, but also dangerous because they may contain links that lead to phishing web sites or sites that are hosting malware - or include malware as file attachments.

## 1.1Defining the Questions

1. Predict if an email is spam or not 




## 1.2 Metrics for success
* Accuracy and FI score will be used todeterrmine efficiency,the highest the score ,the better the model
*different split sizes 80-20,70-30 and 60-40  will be used to determine the best of the three

 

## 1.3Appropriateness of the data

**Dataset 2 link:**
* [link text](https://archive.ics.uci.edu/ml/datasets/Spambase)
* The collection of spam e-mails came from the postmaster and individuals who had filed spam.

* Drawback :This dataset has no column names.



# Data Understanding

## 2.1 Importing libraries

In [4]:
# Importing Libraries we will use for this project analysis

import pandas as pd  # for data processing
import numpy as np  # for numerical calculations


# Data Visualization Libraries
import seaborn as sns
import seaborn as sb

import matplotlib.pyplot as plt

%matplotlib inline



# Sklearn libraries for data preparartion and performance measures
from sklearn.model_selection import train_test_split, RandomizedSearchCV, GridSearchCV, KFold, cross_val_predict
from sklearn.preprocessing import StandardScaler, MinMaxScaler, RobustScaler, normalize, Normalizer
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as LDA
from sklearn.decomposition import PCA


# Algorithms
from sklearn.naive_bayes import MultinomialNB, GaussianNB, BernoulliNB
from sklearn.neighbors import KNeighborsClassifier

from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
from xgboost import XGBClassifier


## 1.2 Loading the datasets

In [5]:
# Loading the dataset

sm= pd.read_csv('spambase.csv')



## 2.3 Checking data

In [19]:
# Viewing the first  five observations of the  dataset
sm.head(5)

0  0.64  0.64.1  0.1  0.32   0.2  ...   0.43   0.44  3.756   61   278  1
0  0.21  0.28    0.50  0.0  0.14  0.28  ...  0.180  0.048  5.114  101  1028  1
1  0.06  0.00    0.71  0.0  1.23  0.19  ...  0.184  0.010  9.821  485  2259  1
2  0.00  0.00    0.00  0.0  0.63  0.00  ...  0.000  0.000  3.537   40   191  1
3  0.00  0.00    0.00  0.0  0.63  0.00  ...  0.000  0.000  3.537   40   191  1
4  0.00  0.00    0.00  0.0  1.85  0.00  ...  0.000  0.000  3.000   15    54  1

[5 rows x 58 columns]

In [4]:
# Viewing the first five observations of the  dataset
sm.head(5)

0  0.64  0.64.1  0.1  0.32   0.2  ...   0.43   0.44  3.756   61   278  1
0  0.21  0.28    0.50  0.0  0.14  0.28  ...  0.180  0.048  5.114  101  1028  1
1  0.06  0.00    0.71  0.0  1.23  0.19  ...  0.184  0.010  9.821  485  2259  1
2  0.00  0.00    0.00  0.0  0.63  0.00  ...  0.000  0.000  3.537   40   191  1
3  0.00  0.00    0.00  0.0  0.63  0.00  ...  0.000  0.000  3.537   40   191  1
4  0.00  0.00    0.00  0.0  1.85  0.00  ...  0.000  0.000  3.000   15    54  1

[5 rows x 58 columns]

In [5]:
# Checking the number of rows and columns in the dataset

# Get the number of rows and columns 
rows = len(sm.axes[0]) 
cols = len(sm.axes[1]) 
  
# Print the number of rows and columns 
print("Number of Rows in this  dataset is: " + str(rows)) 
print("Number of Columns in this  dataset is : " + str(cols)) 


Number of Rows in this  dataset is: 4600
Number of Columns in this  dataset is : 58


#3 Data Preparation

In [6]:
# viewing summary information for our dataset set

sm.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4600 entries, 0 to 4599
Data columns (total 58 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       4600 non-null   float64
 1   0.64    4600 non-null   float64
 2   0.64.1  4600 non-null   float64
 3   0.1     4600 non-null   float64
 4   0.32    4600 non-null   float64
 5   0.2     4600 non-null   float64
 6   0.3     4600 non-null   float64
 7   0.4     4600 non-null   float64
 8   0.5     4600 non-null   float64
 9   0.6     4600 non-null   float64
 10  0.7     4600 non-null   float64
 11  0.64.2  4600 non-null   float64
 12  0.8     4600 non-null   float64
 13  0.9     4600 non-null   float64
 14  0.10    4600 non-null   float64
 15  0.32.1  4600 non-null   float64
 16  0.11    4600 non-null   float64
 17  1.29    4600 non-null   float64
 18  1.93    4600 non-null   float64
 19  0.12    4600 non-null   float64
 20  0.96    4600 non-null   float64
 21  0.13    4600 non-null   float64
 22  

In [7]:
# Summary statistics of the dataset
# 
sm.describe()

0         0.64  ...           278            1
count  4600.000000  4600.000000  ...   4600.000000  4600.000000
mean      0.104576     0.212922  ...    283.290435     0.393913
std       0.305387     1.290700  ...    606.413764     0.488669
min       0.000000     0.000000  ...      1.000000     0.000000
25%       0.000000     0.000000  ...     35.000000     0.000000
50%       0.000000     0.000000  ...     95.000000     0.000000
75%       0.000000     0.000000  ...    265.250000     1.000000
max       4.540000    14.280000  ...  15841.000000     1.000000

[8 rows x 58 columns]

##3.1 Missing values

In [8]:
# Checking for null values

sm.isnull().sum()

0         0
0.64      0
0.64.1    0
0.1       0
0.32      0
0.2       0
0.3       0
0.4       0
0.5       0
0.6       0
0.7       0
0.64.2    0
0.8       0
0.9       0
0.10      0
0.32.1    0
0.11      0
1.29      0
1.93      0
0.12      0
0.96      0
0.13      0
0.14      0
0.15      0
0.16      0
0.17      0
0.18      0
0.19      0
0.20      0
0.21      0
0.22      0
0.23      0
0.24      0
0.25      0
0.26      0
0.27      0
0.28      0
0.29      0
0.30      0
0.31      0
0.32.2    0
0.33      0
0.34      0
0.35      0
0.36      0
0.37      0
0.38      0
0.39      0
0.40      0
0.41      0
0.42      0
0.778     0
0.43      0
0.44      0
3.756     0
61        0
278       0
1         0
dtype: int64

there are no missing values

## 3.2 Duplicates

In [9]:
# checking for duplicates
sm.duplicated().sum()

391

There are 391  duplicates and we will drop them

In [10]:
# Dropping for duplicates 

sm.drop_duplicates(inplace = True)

# Cconfriming that we no longer have duplicates

sm.duplicated().sum()

0

Duplicates have been successfully dropped

#4 Building  Machine Learning Models

## 4.1Naive Bayes Classifier

* The Naive Bayes Classifier is a statistical classification technique based on the Bayes Theorem. 

* It has high accuracy and speed on large datasets.

* This type of classifier takes into account the assumption that the effect of a particular feature in a class is independent of other features.


Multinomial NB considers the frequency count (occurrences) of the features  while Bernoulli NB cares only about the presence or absence of a particular feature (word) in the document. The latter is adequate for features that are binary-valued (Bernoulli, boolean). Whereas, with Gaussian NB, features are real-valued or continuous and their distribution is Gaussian, 


splitting data in 80-20 set

In [13]:
X = sm.iloc[:, 0:-1]
y = sm.iloc[:,-1]



X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.2, random_state = 0)

sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)




lda = LDA(n_components=10)
X_train = lda.fit_transform(X_train, y_train)
X_test = lda.transform(X_test)

gnb = GaussianNB()

model = gnb.fit(X_train, y_train)

y_pred = model.predict(X_test)


# Evaluating our model using accuracy score, confusion matrix and classification report.
print(accuracy_score(y_test, y_pred))

print('\n')

print(confusion_matrix(y_test, y_pred))

print('\n')

print(classification_report(y_test, y_pred))



0.8978622327790974


[[475  20]
 [ 66 281]]


              precision    recall  f1-score   support

           0       0.88      0.96      0.92       495
           1       0.93      0.81      0.87       347

    accuracy                           0.90       842
   macro avg       0.91      0.88      0.89       842
weighted avg       0.90      0.90      0.90       842



/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:463: ChangedBehaviorWarning: n_components cannot be larger than min(n_features, n_classes - 1). Using min(n_features, n_classes - 1) = min(57, 2 - 1) = 1 components.
  ChangedBehaviorWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:469: FutureWarning: In version 0.23, setting n_components > min(n_features, n_classes - 1) will raise a ValueError. You should set n_components to None (default), or a value smaller or equal to min(n_features, n_classes - 1).
  warnings.warn(future_msg, FutureWarning)


* The 80,20 split model yielded 89.78% accuracy.
* Interpreting the confusion matrix;
* The first row is about the non-spam-predictions:

      *475 emails were correctly classified as Ham (called true negatives) 
      * 66 were wrongly classified as ham (false positives).
* The second row is about the spam-predictions: 
     * 20 emails where wrongly classified as spam (false negatives) and
     * 281 were correctly classified as Spam (true positives).
     


Splitting  into 70, 30 sets 

In [23]:
X = sm.iloc[:, 0:-1]
y = sm.iloc[:,-1]



X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.3, random_state = 0)

sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)




lda = LDA(n_components=10)
X_train = lda.fit_transform(X_train, y_train)
X_test = lda.transform(X_test)

gnb = GaussianNB()

model = gnb.fit(X_train, y_train)

y_pred = model.predict(X_test)


# Evaluating our model using accuracy score, confusion matrix and classification report.
print(accuracy_score(y_test, y_pred))

print('\n')

print(confusion_matrix(y_test, y_pred))

print('\n')
print(classification_report(y_test, y_pred))

0.9036231884057971


[[789  33]
 [100 458]]


              precision    recall  f1-score   support

           0       0.89      0.96      0.92       822
           1       0.93      0.82      0.87       558

    accuracy                           0.90      1380
   macro avg       0.91      0.89      0.90      1380
weighted avg       0.91      0.90      0.90      1380



/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:463: ChangedBehaviorWarning: n_components cannot be larger than min(n_features, n_classes - 1). Using min(n_features, n_classes - 1) = min(57, 2 - 1) = 1 components.
  ChangedBehaviorWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:469: FutureWarning: In version 0.23, setting n_components > min(n_features, n_classes - 1) will raise a ValueError. You should set n_components to None (default), or a value smaller or equal to min(n_features, n_classes - 1).
  warnings.warn(future_msg, FutureWarning)


* The 70,30 split model also yielded 90.36% accuracy.(higher than previous model)
* Interpreting the confusion matrix;
* The first row is about the non-spam-predictions:
     * 789 emails were correctly classified as Ham (called true negatives) 
     *100 were wrongly classified as ham (false positives).
* The second row is about the spam-predictions: 
     * 33 emails where wrongly classified as spam (false negatives) and
     * 458 were correctly classified as Spam (true positives).
     

Splitting  into 60, 40 sets 



In [24]:
X = sm.iloc[:, 0:-1]
y = sm.iloc[:,-1]



X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.4, random_state = 0)

sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)




lda = LDA(n_components=10)
X_train = lda.fit_transform(X_train, y_train)
X_test = lda.transform(X_test)

gnb = GaussianNB()

model = gnb.fit(X_train, y_train)

y_pred = model.predict(X_test)


# Evaluating our model using accuracy score, confusion matrix and classification report.
print(accuracy_score(y_test, y_pred))

print('\n')

print(confusion_matrix(y_test, y_pred))

print('\n')
print(classification_report(y_test, y_pred))

0.9081521739130435


[[1060   37]
 [ 132  611]]


              precision    recall  f1-score   support

           0       0.89      0.97      0.93      1097
           1       0.94      0.82      0.88       743

    accuracy                           0.91      1840
   macro avg       0.92      0.89      0.90      1840
weighted avg       0.91      0.91      0.91      1840



/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:463: ChangedBehaviorWarning: n_components cannot be larger than min(n_features, n_classes - 1). Using min(n_features, n_classes - 1) = min(57, 2 - 1) = 1 components.
  ChangedBehaviorWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:469: FutureWarning: In version 0.23, setting n_components > min(n_features, n_classes - 1) will raise a ValueError. You should set n_components to None (default), or a value smaller or equal to min(n_features, n_classes - 1).
  warnings.warn(future_msg, FutureWarning)


* The 60,40 split model yielded 90.81% accuracy(Higest of all the models)

* Interpreting the confusion matrix;
* The first row is about the non-spam-predictions:
     * 1060 emails were correctly classified as Ham (called true negatives) 
     * 132 were wrongly classified as ham (false positives).
* The second row is about the spam-predictions: 
     * 37 emails where wrongly classified as spam (false negatives) and
     * 611 were correctly classified as Spam (true positives)..
     


#5 Challenging the solution

SVM builds a classifier by searching for a separating hyperplane (optimal hyperplane) which is optimal and maximises the margin that separates the categories (in our case spam and ham/not spam). Thus, SVM has the advantage of robustness in general and effectiveness when the number of dimensions is greater than the number of samples.
The 60-40 split set will be used to challenge the nayes bayes as it was the best split set 

In [20]:
# Fitting the Support Vector Classifier
# Splitting the data

X = sm.iloc[:, 0:-1]
y = sm.iloc[:,-1]



X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.4, random_state = 0)

sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)




lda = LDA(n_components=10)
X_train = lda.fit_transform(X_train, y_train)
X_test = lda.transform(X_test)

svc = SVC(C=0.1, gamma=0.01, kernel = 'linear')

model = svc.fit(X_train, y_train)

y_pred = model.predict(X_test)


# Evaluating our model using accuracy score, confusion matrix and classification report.
print(accuracy_score(y_test, y_pred))

print('\n')

print(confusion_matrix(y_test, y_pred))

print('\n')

print(classification_report(y_test, y_pred))

0.9217391304347826


[[1041   56]
 [  88  655]]


              precision    recall  f1-score   support

           0       0.92      0.95      0.94      1097
           1       0.92      0.88      0.90       743

    accuracy                           0.92      1840
   macro avg       0.92      0.92      0.92      1840
weighted avg       0.92      0.92      0.92      1840



/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:463: ChangedBehaviorWarning: n_components cannot be larger than min(n_features, n_classes - 1). Using min(n_features, n_classes - 1) = min(57, 2 - 1) = 1 components.
  ChangedBehaviorWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:469: FutureWarning: In version 0.23, setting n_components > min(n_features, n_classes - 1) will raise a ValueError. You should set n_components to None (default), or a value smaller or equal to min(n_features, n_classes - 1).
  warnings.warn(future_msg, FutureWarning)


In [26]:
# Fitting the Support Vector Classifier
# Splitting the data

X = sm.iloc[:, 0:-1]
y = sm.iloc[:,-1]



X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.4, random_state = 0)

sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)



lda = LDA(n_components=10)
X_train = lda.fit_transform(X_train, y_train)
X_test = lda.transform(X_test)

svc = SVC(C=0.1, gamma=0.01, kernel = 'rbf')

model = svc.fit(X_train, y_train)

y_pred = model.predict(X_test)


# Evaluating our model using accuracy score, confusion matrix and classification report.
print(accuracy_score(y_test, y_pred))

print('\n')

print(confusion_matrix(y_test, y_pred))

print('\n')

print(classification_report(y_test, y_pred))

/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:463: ChangedBehaviorWarning: n_components cannot be larger than min(n_features, n_classes - 1). Using min(n_features, n_classes - 1) = min(57, 2 - 1) = 1 components.
  ChangedBehaviorWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:469: FutureWarning: In version 0.23, setting n_components > min(n_features, n_classes - 1) will raise a ValueError. You should set n_components to None (default), or a value smaller or equal to min(n_features, n_classes - 1).
  warnings.warn(future_msg, FutureWarning)


0.903804347826087


[[1061   36]
 [ 141  602]]


              precision    recall  f1-score   support

           0       0.88      0.97      0.92      1097
           1       0.94      0.81      0.87       743

    accuracy                           0.90      1840
   macro avg       0.91      0.89      0.90      1840
weighted avg       0.91      0.90      0.90      1840



In [27]:
# Fitting the Support Vector Classifier
# Splitting the data

X = sm.iloc[:, 0:-1]
y = sm.iloc[:,-1]



X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.4, random_state = 0)

sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)


lda = LDA(n_components=10)
X_train = lda.fit_transform(X_train, y_train)
X_test = lda.transform(X_test)

svc = SVC(C=0.1, gamma=0.01, kernel = 'poly')

model = svc.fit(X_train, y_train)

y_pred = model.predict(X_test)


# Evaluating our model using accuracy score, confusion matrix and classification report.
print(accuracy_score(y_test, y_pred))

print('\n')

print(confusion_matrix(y_test, y_pred))

print('\n')

print(classification_report(y_test, y_pred))

/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:463: ChangedBehaviorWarning: n_components cannot be larger than min(n_features, n_classes - 1). Using min(n_features, n_classes - 1) = min(57, 2 - 1) = 1 components.
  ChangedBehaviorWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:469: FutureWarning: In version 0.23, setting n_components > min(n_features, n_classes - 1) will raise a ValueError. You should set n_components to None (default), or a value smaller or equal to min(n_features, n_classes - 1).
  warnings.warn(future_msg, FutureWarning)


0.5967391304347827


[[1097    0]
 [ 742    1]]


              precision    recall  f1-score   support

           0       0.60      1.00      0.75      1097
           1       1.00      0.00      0.00       743

    accuracy                           0.60      1840
   macro avg       0.80      0.50      0.37      1840
weighted avg       0.76      0.60      0.45      1840



In [28]:
# Fitting the Support Vector Classifier
# Splitting the data

X = sm.iloc[:, 0:-1]
y = sm.iloc[:,-1]



X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.4, random_state = 0)

sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)



lda = LDA(n_components=10)
X_train = lda.fit_transform(X_train, y_train)
X_test = lda.transform(X_test)

svc = SVC(C=0.1, gamma=0.01, kernel = 'sigmoid')

model = svc.fit(X_train, y_train)

y_pred = model.predict(X_test)


# Evaluating our model using accuracy score, confusion matrix and classification report.
print(accuracy_score(y_test, y_pred))

print('\n')

print(confusion_matrix(y_test, y_pred))

print('\n')

print(classification_report(y_test, y_pred))

/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:463: ChangedBehaviorWarning: n_components cannot be larger than min(n_features, n_classes - 1). Using min(n_features, n_classes - 1) = min(57, 2 - 1) = 1 components.
  ChangedBehaviorWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:469: FutureWarning: In version 0.23, setting n_components > min(n_features, n_classes - 1) will raise a ValueError. You should set n_components to None (default), or a value smaller or equal to min(n_features, n_classes - 1).
  warnings.warn(future_msg, FutureWarning)


0.883695652173913


[[1067   30]
 [ 184  559]]


              precision    recall  f1-score   support

           0       0.85      0.97      0.91      1097
           1       0.95      0.75      0.84       743

    accuracy                           0.88      1840
   macro avg       0.90      0.86      0.87      1840
weighted avg       0.89      0.88      0.88      1840



In [33]:
#e linear kernel perform best for this partical classification problem.

# Here we will use Grid search to find the best parameters for the model

# Creating a dictionary
# Specifying the parameters we want to tune

params = {'C': [0.00001,0.001,0.1],
          'gamma': [0.01, 0.001,0.0001]}

svc = SVC()

search = GridSearchCV(estimator = svc, 
                     param_grid= params, 
                      scoring = 'accuracy',
                      cv = 10 )
                
                
search.fit(X,y)
search.best_params_

{'C': 0.1, 'gamma': 0.001}

In [38]:
# Fitting the Support Vector Classifier
# Splitting the data

X = sm.iloc[:, 0:-1]
y = sm.iloc[:,-1]



X_train, X_test, y_train, y_test = train_test_split(X,y, test_size = 0.4, random_state = 0)

sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)




lda = LDA(n_components=10)
X_train = lda.fit_transform(X_train, y_train)
X_test = lda.transform(X_test)

svc = SVC(C=0.1, gamma=0.001, kernel = 'linear')

model = svc.fit(X_train, y_train)

y_pred = model.predict(X_test)


# Evaluating our model using accuracy score, confusion matrix and classification report.
print(accuracy_score(y_test, y_pred))

print('\n')

print(confusion_matrix(y_test, y_pred))

print('\n')

print(classification_report(y_test, y_pred))

0.9217391304347826


[[1041   56]
 [  88  655]]


              precision    recall  f1-score   support

           0       0.92      0.95      0.94      1097
           1       0.92      0.88      0.90       743

    accuracy                           0.92      1840
   macro avg       0.92      0.92      0.92      1840
weighted avg       0.92      0.92      0.92      1840



/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:463: ChangedBehaviorWarning: n_components cannot be larger than min(n_features, n_classes - 1). Using min(n_features, n_classes - 1) = min(57, 2 - 1) = 1 components.
  ChangedBehaviorWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/discriminant_analysis.py:469: FutureWarning: In version 0.23, setting n_components > min(n_features, n_classes - 1) will raise a ValueError. You should set n_components to None (default), or a value smaller or equal to min(n_features, n_classes - 1).
  warnings.warn(future_msg, FutureWarning)


#6 Conclusion

* SVM performs better than naives bayes in this project had highest score
 at 92.17% when the kernel is linear
* The best split size was 60-40
*Hypertuning svm gave me the same result before meaning the model was performing optimally already